packages/component/src/hooks/useRenderEmoji.js

```diff
@@ -0,0 +1,23 @@
+import useEmojiFromStyles from './useEmojiFromStyles';
+import { useTextBoxValue } from '../SendBox/TextBox';
+import { useCallback } from 'react';
+
+export default function useRenderEmoji() {
+  const [{ emojiAutocorrect, emojiUnicodeMap, emojiRegExp }] = useEmojiFromStyles();
+
+  const [value] = useTextBoxValue();
+
+  const setEmoji = useCallback(
+    value => {
+      const emoticon = value.match(emojiRegExp);
+
+      if (emojiAutocorrect && !!emoticon && !!emojiUnicodeMap[emoticon]) {
+        value = value.replace(emoticon, emojiUnicodeMap[emoticon]);
+      }
+      return value;
+    },
+    [emojiAutocorrect, emojiUnicodeMap, emojiRegExp]
+  );
+
+  return [value, setEmoji];
+}
```

Comment:
1. Please add a sample on how to customize the emoji set.
1. Did you test pressing <kbd>CTRL</kbd> + <kbd>Z</kbd> after an emoticon is replaced with emoji? Can you add a test?
   - <kbd>CTRL</kbd> + <kbd>Z</kbd> may not work, but it should not break the UX.
1. Add a test for pasting text via <kbd>CTRL</kbd> + <kbd>V</kbd>. And set an expectation.
   - Expectations means: pasted text will have emoticons converted, or pasted text will not have emoticons converted
   - Please paste a text with 2+ emoticons

> FYI, some tests can only run under WebDriver but not local Chrome because <kbd>CTRL</kbd> + <kbd>V</kbd>/<kbd>Z</kbd> cannot be emulated using JS.